src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets

```diff
@@ -141,34 +186,36 @@ Copyright (c) .NET Foundation. All rights reserved.
 </Project>        
 
 ]]>
-      </_CrossgenProjContent>
+      </CrossgenProjContent>
     </PropertyGroup>
 
     <PropertyGroup>
-      <_RuntimeDir>$([System.IO.Path]::Combine($(_CrossProjFileDir),"runtime"))</_RuntimeDir>   <!-- app managed assemblies in nuget cache layout-->
-      <_RuntimeRefDir>$([System.IO.Path]::Combine($(_CrossProjFileDir), "runtimeref"))</_RuntimeRefDir> <!-- flat app managed assemblies -->
-      <_NetCoreRefDir>$([System.IO.Path]::Combine($(_CrossProjFileDir), "netcoreapp"))</_NetCoreRefDir> <!-- flat netcore app assemblies-->
-      <_RuntimeOptimizedDir>$([System.IO.Path]::Combine($(_CrossProjFileDir), "runtimopt"))</_RuntimeOptimizedDir> <!-- optimized app managed assemblies in nuget cache layout -->
-      <_CrossProjFile>$([System.IO.Path]::Combine($(_CrossProjFileDir), "Optimize_Crossgen_$(RuntimeIdentifier)_$(_TFM).csproj"))</_CrossProjFile>
+      <!-- optimized app managed assemblies in nuget cache layout -->
+      <_CrossProjFile>$([System.IO.Path]::Combine($(_CrossProjFileDir), "OptimizeCrossgen_$(RuntimeIdentifier)_$(_TFM).csproj"))</_CrossProjFile>
       <_CrossProjAssetsFile>$([System.IO.Path]::Combine($(_CrossProjFileDir), "obj", project.assets.json))</_CrossProjAssetsFile>
     </PropertyGroup>
 
     <MakeDir  Directories="$(_CrossProjFileDir)"/>
-    <MakeDir  Directories="$(_RuntimeDir)"/>
+    
     <WriteLinesToFile
         File="$(_CrossProjFile)"
-        Lines="$(_CrossgenProjContent)"
+        Lines="$(CrossgenProjContent)"
         Overwrite="true" />
   </Target>
 
   <!--
     ============================================================
-                                        _SetupCrossgenStage
+                                        _SetupStageForCrossgen
     ============================================================
     -->
-  <Target Name="_SetupCrossgenStage"
-          DependsOnTargets="_FilterOutFXFromResolvedCandidates;_PoulateUnFilteredResolvedCandidates">
-
+  <Target Name="_SetupStageForCrossgen"
+          DependsOnTargets="_GetCrossgenProps;_FilterOutFXFromResolvedCandidates;_PoulateUnFilteredResolvedCandidates">
+    <PropertyGroup>
+      <_RuntimeDir>$([System.IO.Path]::Combine($(CacheWorkerWorkingDir),"runtime"))</_RuntimeDir>            <!-- app managed assemblies in nuget cache layout-->
```

Comment:
No one else does this anywhere in the product.  This code should follow the same patterns used everywhere else.